dev-requirements.txt

```diff
@@ -1,18 +1,22 @@
-mock==1.3.0
-coverage==3.7.1
+mock==2.0.0
+coverage==4.5.1
 tox==2.9.1
-twine==1.5.0
+twine==1.11.0
 wheel==0.30.0
-tornado==4.2.1
-PySocks==1.5.6
-pkginfo>=1.0,!=1.3.0
-psutil==4.3.1
+tornado==4.5.3
+PySocks==1.6.8
+pkginfo==1.4.2
+psutil==5.4.3
 pytest-cov==2.5.1
-pytest-timeout==1.2.0
-pytest==3.4.0
-gcp-devrel-py-tools==0.0.7
+pytest-random-order==0.6.0
```

Comment:
I originally added this to help track down the weird timeout bug we were hitting on Windows, and then forgot to remove it. But maybe it's useful to keep -- what do you think?

Reply:
I do like it! It can be surprising/annoying at times, but I've seen enough tests depending on each other to know that it can help detect annoying issues earlier.

Reply:
However it seems to break --failed-first, which is possibly more useful.